distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex/Match_2.enso

```diff
@@ -271,7 +271,7 @@ type Match_2
        Arguments:
        - id: The integer index or name of that group.
        - if_missing: The value to return if the index is out of bounds.
-    get : Integer -> Any -> Any
+    get : Integer -> Any -> Text
     get self index ~if_missing=Nothing =
         self.text index . catch No_Such_Group (_-> if_missing)
```

Comment:
I'm wondering how to handle this.

Most of our collections are holding `Any` anyway, so it was not an issue. But here it shows up.

`get` will indeed return the `Text` type if we find the given group.
But if we don't find it, we return `if_missing`, which can technically have _any_ type.
So that would suggest the signature was correct before the change.

OTOH, I think the new signature is more intuitive, as the idea behind this function is that it should normally return `Text`. So I'm not saying to change it back, really I'm not sure what is the right choice here.

Most correct could be if we had polymorphic types:
```
get : Integer -> a -> Text | a
```
which would correctly express that it's either a `Text` or the thing of the type that was provided to `if_missing` (being `Nothing` by default, but possibly overridden to sth else, changing the type).

But currently we are not writing polymorphic signatures like this (probably cannot do that).

Reply:
I suggest go with the slightly weird `Text | Any` for now.